django/chat/responses.py

```diff
@@ -159,15 +159,16 @@ def summarize_response(chat, response_message):
         for file in files:
             if not file.text:
                 file.extract_text(pdf_method="default")
-            responses.append(
-                summarize_long_text(
-                    file.text,
-                    llm,
-                    summary_length,
-                    target_language,
-                    custom_summarize_prompt,
+            if not cache.get(f"stop_response_{response_message.id}", False):
```

Comment:
Due to the way the summarize_response logic works, it's still possible to prevent summarization outright by hitting the stop button reasonably quickly after the uploads. In this case, the output will be blank and there will be no cost object created - but I've verified that it doesn't actually go through to the model.

If you're _not_ quick enough, you get the "Response stopped" message and whatever costs accumulated from the summarization jobs that went through, as expected.